src/pages/VeSDL/OnChainVote.tsx

```diff
@@ -0,0 +1,254 @@
+        }
+      } catch (error) {
+        console.log("error on vote ==>", error)
+      }
+    }
+  }
+
+  return (
+    <div>
+      <Stack spacing={2} px={2} pt={2} mb={2}>
+        <Typography variant="h2" textAlign="center">
+          {t("voteForNextWeek")}
+        </Typography>
+        <Box display="flex" justifyContent="space-between">
+          <Box display="flex">
+            <Typography component="label">
+              {t("yourVotingPower")}:{" "}
+              <Typography component="span" mx="8px">
+                {commify(formatBNToString(veSdlBalance, 18, 2))}
+              </Typography>
+              veSDL
+            </Typography>
+          </Box>
+          <Link
+            sx={{
+              color: theme.palette.text.primary,
+              textDecorationColor: theme.palette.text.primary,
+              whiteSpace: "nowrap",
+            }}
+          >
+            {t("gaugeDoc")}
+          </Link>
+        </Box>
+        {alertMessage && (
+          <Alert severity="error">
+            <Typography textAlign="center">{alertMessage}</Typography>
+          </Alert>
+        )}
+        <Box
+          border={`1px solid ${theme.palette.other.divider}`}
+          px={2}
+          pt={2}
+          borderRadius="10px"
+        >
+          <Autocomplete
+            id="gauge-names"
+            options={gaugeNames}
+            getOptionLabel={(option) => option?.gaugeName ?? ""}
+            popupIcon={<ArrowDownIcon />}
+            renderInput={(params) => (
+              <TextField
+                variant="standard"
+                {...params}
+                placeholder={t("chooseGauge")}
+              />
+            )}
+            onChange={(event, newGaugeName) => {
+              if (newGaugeName) {
+                setSelectedGauge(newGaugeName)
+              }
+            }}
+            //value of selected gauge
+            value={selectedGauge}
+            // check whether the current value belong to options
+            isOptionEqualToValue={(option, value) =>
+              option?.address === value?.address
+            }
+          />
+          <Box display="flex" gap={4} my={2}>
+            <TextField
+              variant="standard"
+              label={t("votingWeight")}
+              type="text"
+              value={voteWeightToSubmit}
+              onChange={(event) => {
+                if (isNumberOrEmpty(event.target.value)) {
+                  const estimatedWeight =
+                    parseFloat(event.target.value) + (voteUsed ?? 0) / 100
+                  if (estimatedWeight > 100) {
+                    setVoteWeightInputError({
+                      hasError: true,
+                      errorText: "Used too much power",
+                    })
+                  } else {
+                    setVoteWeightInputError({
+                      hasError: false,
+                      errorText: " ",
```

Comment:
Can you add a reasonable error message here?

Reply:
We should set `space` here because we don't need to display message if estimated weight is less than `100%`